s3fs/tests/test_s3fs.py

```diff
@@ -25,21 +26,103 @@
                                 b'Dennis,400,4\n'
                                 b'Edith,500,5\n'
                                 b'Frank,600,6\n')}
+text_files = {'nested/file1': b'hello\n',
+              'nested/file2': b'world',
+              'nested/nested2/file1': b'hello\n',
+              'nested/nested2/file2': b'world'}
+a = test_bucket_name+'/tmp/test/a'
+b = test_bucket_name+'/tmp/test/b'
+c = test_bucket_name+'/tmp/test/c'
+d = test_bucket_name+'/tmp/test/d'
+
 
 @pytest.yield_fixture
 def s3():
-    # could do with a bucket with write privileges.
-    yield S3FileSystem(anon=True)
-
-
-def test_non_anonymous_access():
+    # writable local S3 system
+    m = moto.mock_s3()
+    m.start()
+    import boto3
+    client = boto3.client('s3')
+    client.create_bucket(Bucket=test_bucket_name)
+    for k in [a, b, c, d]:
+        try:
+            client.delete_object(Bucket=test_bucket_name, Key=k)
```

Comment:
If the object didn't exist - i.e., a failing earlier test may or may not pollute the bucket with leftover stuff and cause other tests to fail